src/wham_v0.cpp

```diff
@@ -465,6 +591,847 @@ Type objective_function<Type>::operator() ()
+			}
+		}
+	}
+  }
+  
+  // add ecov effect on growth paramteres
+  int count_wheresub = 0; // used for growth and LW
+  for(int j = 0; j < n_growth_par; j++) { 
+	for(int i=0; i < n_Ecov; i++){
+		if((Ecov_where(i,n_effects-4) == 1) & ((Ecov_where_subindex(count_wheresub)-1) == j)) {  // for growth
+			for(int y = 0; y < n_years_model + n_years_proj; y++) {
+				for(int a = 0; a < n_ages; a++) { 
+					GW_par(y,a,j) *= exp(Ecov_lm(i,n_effects-4,y,0)); 
+				}
+			}
+			count_wheresub += 1; 
+		}
+	}
+  }
+
+  // add ecov effect on LAA
+	for(int i=0; i < n_Ecov; i++){
+		if((Ecov_where(i,n_effects-3) == 1)) { 
+			for(int y = 0; y < n_years_model + n_years_proj; y++) {
+				for(int a = 0; a < n_ages; a++) { 
+					LAA_par(y,a) *= exp(Ecov_lm(i,n_effects-3,y,a));
+				}
+			}
+		}
+	}
+	
+  // Update SD_par:
+  vector<Type> expSD(2); // always 2 parameters
+  for(int i=0; i < 2; i++) {
+	  if(growth_model < 3) expSD(i) = exp(SD_par(i)); 
+	  else expSD(i) = exp(SDLAA_par(i)); // nonparametric approach
+  }
+  REPORT(expSD);
+
+  // ---------------------------------------------------------------------
+  // Construct WAA parameters per year
+  matrix<Type> WAA_par(n_years_model + n_years_proj,n_ages); 
+  for(int y = 0; y < n_years_model; y++) { 
+		for(int a = 0; a < n_ages; a++) { 
+			WAA_par(y,a) = exp(WAA_a(a) + WAA_re(y,a)); 
+		}
+  }
+
+
+  // add to WAA parameters in projection years
+  if((do_proj == 1) & (weight_model == 3)){ 
+	if(proj_WAA_opt(0) == 2){
+	  matrix<Type> WAA_toavg(n_toavg,n_ages);
+	  for(int a = 0; a < n_ages; a++){
+		for(int i = 0; i < n_toavg; i++){
+		  WAA_toavg(i,a) = WAA_par(avg_years_ind(i),a);
+		}
+	  }
+	  vector<Type> WAA_proj = WAA_toavg.colwise().mean();
+	  for(int y = n_years_model; y < n_years_model + n_years_proj; y++){
+		WAA_par.row(y) = WAA_proj;
+	  }
+	} else { // proj_WAA_opt == 1
+		for(int y = n_years_model; y < n_years_model + n_years_proj; y++) {
+			for(int a = 0; a < n_ages; a++) { 		
+				WAA_par(y,a) = exp(WAA_a(a) + WAA_re(y,a)); 
+			}
+		}
+	}
+  }
+  
+  // add ecov effect on WAA
+	for(int i=0; i < n_Ecov; i++){
+		if(Ecov_where(i,n_effects-1) == 1) {  
+			for(int y = 0; y < n_years_model + n_years_proj; y++) {
+				for(int a = 0; a < n_ages; a++) { 
+					WAA_par(y,a) *= exp(Ecov_lm(i,n_effects-1,y,a));
+				}
+			}
+		}
+    }
+
+
+
+  // --------------------------------------------------------------------------
+  // Calculate LW -- 
+  Type nll_LW = Type(0);
+  
+  // For all growth parameters:
+  vector<Type> sigma_LW(n_LW_par); // first RE parameter
+  vector<Type> rho_LW_y(n_LW_par);  // second RE parameter
+
+  for(int j = 0; j < n_LW_par; j++) {
+  
+  Type Sigma_LW = Type(0);
+
+	  if((LW_re_model(j) == 2) | (LW_re_model(j) == 4)) { // Only for ii_y and Ar1_y. CHECK THIS LATER, ONLY FOR WORK AR_y so far I think
+		
+		sigma_LW(j) = exp(LW_repars(j,0));
+        rho_LW_y(j) = rho_trans(LW_repars(j,1));  
+		// likelihood of growth parameters deviations
```

Comment:
Looks like you copied growth code and forgot to update the comments to LW throughout here

Reply:
done!